JOSESwift/Sources/Decrypter.swift

```diff
@@ -135,15 +136,15 @@ public struct Decrypter {
 
             cek = symmetricKey
         } else {
-            // Generate random CEK to prevent MMA (Million Message Attack).
-            // For detailed information, please refer to this RFC(https://tools.ietf.org/html/rfc3218#section-2.3.2)
-            // and http://www.ietf.org/mail-archive/web/jose/current/msg01832.html
-            let randomCEK = try SecureRandom.generate(count: enc.keyLength)
-
-            if let decryptedCEK = try? asymmetric.decrypt(context.encryptedKey) {
-                cek = decryptedCEK
-            } else {
-                cek = randomCEK
+        // Generate random CEK to prevent MMA (Million Message Attack).
+        // For detailed information, please refer to this RFC(https://tools.ietf.org/html/rfc3218#section-2.3.2)
+        // and http://www.ietf.org/mail-archive/web/jose/current/msg01832.html
+        let randomCEK = try SecureRandom.generate(count: enc.keyLength)
```

Comment:
What do you think of moving this directly into the else block below?

Reply:
As discussed offline, I updated the documentation referencing more resources and examples of where the random CEK to substitute in case of decryption failure is generated before the decryption ([RFC-5246](https://tools.ietf.org/html/rfc5246#appendix-F.1.1.2)). For further reference, see [Nimbus JOSE +JWT](https://bitbucket.org/connect2id/nimbus-jose-jwt/src/a97f123a1ad29906befcb94f9c86a98d47c37fd4/src/main/java/com/nimbusds/jose/crypto/RSADecrypter.java?at=master&fileviewer=file-view-default#RSADecrypter.java-253).

Thanks again for challenging this non-obvious pattern. 👍